### PR TITLE
Add RescanWhileNotMoving to AttackFollow

### PR DESCRIPTION
**Pull request by jrb0001**
This is needed to stop following the current target if it isn't targetable any more. I made it optional because it isn't needed in most cases and has a potential performance hit.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace OpenRA.Mods.Common.Traits
 	[Desc("Actor will follow units until in range to attack them.")]
 	public class AttackFollowInfo : AttackBaseInfo
 	{
+		[Desc("Rescanning detects changes in the targets Targetable traits but has a performance hit.")]
+		public readonly bool RescanWhileNotMoving = false;
```
Comment (abc013) [inline OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: `RescanWhileMoving` would be better i think.

Comment (jrb0001) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: `RescanWhileMoving` would require a rework of `MoveWithinRange` and have a way bigger performance hit.

**Comment by jrb0001** [comment]
As discussed with pchote in [irc](http://logs.openra.net/index.php?year=2017&month=09&day=10#18:00:31):
> ‎<‎pchote‎>‎ AttackMoveActivity is probably the simplest example of what i'm suggesting
> ‎<‎jrb0001‎>‎ pchote: so you mean to use ActivityUtils.RunActivity instead of ActivityUtils.SequenceActivities and build around that?
> ‎<‎pchote‎>‎ jrb0001: not quite "just" but yes
> ‎<‎jrb0001‎>‎ that sounds like good idea although it will have a performance hit if the target can't become invalid (which is the case in (nearly) all current mods afaik).
> ‎<‎pchote‎>‎ shouldn't be *that* significant
> ‎<‎jrb0001‎>‎ I guess it should be enough to make the rescan interval configurable with a high default value.

File: OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs (hunk 1 above)
Comment (abc013) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: I mean renaming `RescanWhileNotMoving` to `RescanWhileMoving` and inverting the check and the initial value.

Comment (jrb0001) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: There is a big difference between not rescanning at all (=current behaviour), rescanning only while not moving (= PR behaviour enabled) and rescanning only while moving (=your proposal enabled).

Anyways, I will rework the PR in the near future which should make rescanning independent from moving.

Comment (abc013) [reply OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs]: Ahh, thanks ;).

**Comment by pchote** [comment]
Any progress here @jrb0001? Or do you want to close this until you are ready?

**Comment by jrb0001** [comment]
I won't work on this PR in the near future.

**Pull request closed without merging**